devito/ir/equations/equation.py

```diff
@@ -110,32 +112,27 @@ def __new__(cls, *args, **kwargs):
                              "and kwargs=%s" % (str(args), str(kwargs)))
 
         # Well-defined dimension ordering
-        ordering = dimension_sort(expr, key=lambda i: not i.is_Time)
```

Comment:
note for reviewers: Dropping this ``lambda i: not i.is_Time`` is another good thing of this PR. If it wasn't a hack, it was very close to it.